ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/forms/CollectionSelection.tsx

```diff
@@ -0,0 +1,207 @@
+import React, { useMemo, useState, ReactElement, useCallback, useEffect } from 'react';
+import {
+    Button,
+    ButtonVariant,
+    Flex,
+    FlexItem,
+    Select,
+    SelectOption,
+    SelectProps,
+    SelectVariant,
+    ValidatedOptions,
+} from '@patternfly/react-core';
+import sortBy from 'lodash/sortBy';
+import uniqBy from 'lodash/uniqBy';
+
+import FormLabelGroup from 'Components/PatternFly/FormLabelGroup';
+import { Collection, listCollections } from 'services/CollectionsService';
+import CollectionsFormModal from 'Containers/Collections/CollectionFormModal';
+import { useCollectionFormSubmission } from 'Containers/Collections/hooks/useCollectionFormSubmission';
+import useSelectToggle from 'hooks/patternfly/useSelectToggle';
+import { usePaginatedQuery } from 'hooks/usePaginatedQuery';
+import { ReportScope } from 'hooks/useFetchReport';
+
+const COLLECTION_PAGE_SIZE = 10;
+
+type CollectionSelectionProps = {
+    selectedScopeId: string;
+    initialReportScope: ReportScope | null;
+    onChange: (selection: string) => void;
+    allowCreate: boolean;
+};
+
+function CollectionSelection({
+    selectedScopeId,
+    initialReportScope,
+    onChange,
+    allowCreate,
+}: CollectionSelectionProps): ReactElement {
+    const { isOpen, onToggle } = useSelectToggle();
+    const { configError, setConfigError, onSubmit } = useCollectionFormSubmission({
+        type: 'create',
+    });
+    const [search, setSearch] = useState('');
+
+    const [isCollectionModalOpen, setIsCollectionModalOpen] = useState(false);
+
+    const requestFn = useCallback(
+        (page: number) => {
+            return listCollections(
+                { 'Collection Name': search },
+                { field: 'Collection Name', reversed: false },
+                page,
+                COLLECTION_PAGE_SIZE
+            ).request;
+        },
+        [search]
+    );
+
+    const { data, isEndOfResults, isFetchingNextPage, fetchNextPage } = usePaginatedQuery(
+        requestFn,
+        COLLECTION_PAGE_SIZE
+    );
+
+    const isLegacyReportScopeSelected =
+        initialReportScope?.type === 'AccessControlScope' &&
+        initialReportScope?.id === selectedScopeId;
```

Comment:
TODO: determine if we can remove conditions that check for Access Control scopes, because those were migrated several releases ago. We shouldn't need to support them in VM 2.0 reports.

cc: @md2119 @dvail

Reply:
Agreed - pretty sure this can be removed. [Here](https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/hooks/useFetchReport.ts#L8) is where the type is declared at the top level, with a comment that we can ditch these post 3.74.